screens/AuthLoadingScreen.js

```diff
@@ -1,26 +1,61 @@
 import React from 'react';
+import { connect } from 'react-redux';
 import {
     ActivityIndicator,
     AsyncStorage,
     StatusBar,
-    StyleSheet,
     View
 } from 'react-native';
+import { Toast } from 'native-base';
 
-export default class AuthLoadingScreen extends React.Component {
+class AuthLoadingScreen extends React.Component {
     constructor(props) {
         super(props);
+
+        this.state = {
+            socket: this.props.socket,
+            navigation: this.props.navigation
+        }
+
         this._bootstrapAsync();
     }
 
+    clearAsyncStorage = async () => {
+        await AsyncStorage.clear();
+    }
+
     _bootstrapAsync = async () => {
+        const { socket, navigation } = this.state;
```

Comment:
Now handling the auth stuff in the auth loading screen. On second thought, I think it'd be better to render some page with the error message instead of presenting a toast here. Maybe that's a change we can introduce later, though...